x/incentives/keeper/distribute_test.go

```diff
@@ -68,22 +67,101 @@ func (suite *KeeperTestSuite) TestDistribute() {
+	}
+	doubleLengthGauge := perpGaugeDesc{
+		lockDenom:    defaultLPSyntheticDenom,
+		lockDuration: 2 * defaultLockDuration,
+		rewardAmount: sdk.Coins{sdk.NewInt64Coin(defaultRewardDenom, 3000)},
+	}
+	noRewardGauge := perpGaugeDesc{
+		lockDenom:    defaultLPSyntheticDenom,
+		lockDuration: defaultLockDuration,
+		rewardAmount: sdk.Coins{},
+	}
+	noRewardCoins := sdk.Coins{}
+	oneKRewardCoins := sdk.Coins{sdk.NewInt64Coin(defaultRewardDenom, 1000)}
+	twoKRewardCoins := sdk.Coins{sdk.NewInt64Coin(defaultRewardDenom, 2000)}
+	fiveKRewardCoins := sdk.Coins{sdk.NewInt64Coin(defaultRewardDenom, 5000)}
+	tests := []struct {
+		name            string
+		users           []userLocks
+		gauges          []perpGaugeDesc
+		expectedRewards []sdk.Coins
+	}{
+		// gauge 1 gives 3k coins. three locks, all eligible. 1k coins per lock.
+		// 1k should go to oneLockupUser and 2k to twoLockupUser.
+		{
+			name:            "test1",
+			users:           []userLocks{oneSyntheticLockupUser, twoSyntheticLockupUser},
+			gauges:          []perpGaugeDesc{defaultGauge},
+			expectedRewards: []sdk.Coins{oneKRewardCoins, twoKRewardCoins},
+		},
+		// gauge 1 gives 3k coins. three locks, all eligible.
+		// gauge 2 gives 3k coins. one lock, to twoLockupUser.
+		// 1k should to oneLockupUser and 5k to twoLockupUser.
+		{
+			name:            "test2",
+			users:           []userLocks{oneSyntheticLockupUser, twoSyntheticLockupUser},
+			gauges:          []perpGaugeDesc{defaultGauge, doubleLengthGauge},
+			expectedRewards: []sdk.Coins{oneKRewardCoins, fiveKRewardCoins},
+		},
+		{
+			name:            "test3",
+			users:           []userLocks{oneSyntheticLockupUser, twoSyntheticLockupUser},
+			gauges:          []perpGaugeDesc{noRewardGauge},
+			expectedRewards: []sdk.Coins{noRewardCoins, noRewardCoins},
+		},
+		{
+			name:            "test4",
+			users:           []userLocks{oneSyntheticLockupUser, twoSyntheticLockupUser},
+			gauges:          []perpGaugeDesc{noRewardGauge, defaultGauge},
+			expectedRewards: []sdk.Coins{oneKRewardCoins, twoKRewardCoins},
+		},
+	}
+	for _, tc := range tests {
+		suite.SetupTest()
+		gauges := suite.SetupGauges(tc.gauges, defaultLPSyntheticDenom)
+		addrs := suite.SetupUserSyntheticLocks(tc.users)
+		_, err := suite.App.IncentivesKeeper.Distribute(suite.Ctx, gauges)
+		suite.Require().NoError(err)
+		// check expected rewards
+		for i, addr := range addrs {
+			var rewards string
+			bal := suite.App.BankKeeper.GetAllBalances(suite.Ctx, addr)
+			// extract the superbonding tokens from the rewards distribution check
+			// figure out a less hacky way of doing this
+			if strings.Contains(bal.String(), "lptoken/superbonding,") {
+				rewards = strings.Split(bal.String(), "lptoken/superbonding,")[1]
+			}
+			suite.Require().Equal(tc.expectedRewards[i].String(), rewards, "test %v, person %d", tc.name, i)
+		}
+	}
 
-	// TODO: test distribution for synthetic lockup as well
```

Comment:
https://github.com/osmosis-labs/osmosis/pull/1885/commits/a0cc53b8504a205919a20d0879aef31d89091f35

My attempt at testing distribution for synthetic lockups